Inferences/Induction.cpp

```diff
@@ -268,41 +205,46 @@ TermList LiteralSubsetReplacement::transformSubterm(TermList trm)
   return trm;
 }
 
-Literal* LiteralSubsetReplacement::transformSubset(InferenceRule& rule) {
-  CALL("LiteralSubsetReplacement::transformSubset");
+bool ContextSubsetReplacement::hasNext()
+{
+  CALL("ContextSubsetReplacement::hasNext");
+  if (_ready) {
+    return hasNextInner();
+  }
+  _ready = true;
   // Increment _iteration, since it either is 0, or was already used.
   _iteration++;
   // Note: __builtin_popcount() is a GCC built-in function.
```

Comment:
Apparently this PR didn't introduce it, but maybe we should try and avoid this kind of thing. If we have a reasonably clever compiler and ask it to do platform-specific stuff it will apparently recognize this anyhow! https://lemire.me/blog/2016/05/23/the-surprising-cleverness-of-modern-compilers/

Reply:
That's nice! So should we just iterate over it ourselves?

Reply:
I think so! I'll factor it out later (probably into `Lib` somewhere) and either use some preprocessor guards or rely on the compiler.